develop/interoperability/xcm-runtime-apis.md

```diff
@@ -0,0 +1,449 @@
+- Exposing formulas for complex computational calculations
+
+In this section, you will learn about specific runtime APIs that support XCM processing and manipulation.
+
+## Dry Run API
+
+The [Dry-run API](https://paritytech.github.io/polkadot-sdk/master/xcm_runtime_apis/dry_run/trait.DryRunApi.html){target=\_blank}, given an extrinsic, or an XCM program, returns its effects:
+
+- Execution result
+- Local XCM (in the case of an extrinsic)
+- Forwarded XCMs
+- List of events
+
+This API can be used on its own for dry-running purposes, for double-checking or testing, but it mainly shines when used in conjunction with the XcmPaymentApi, given that it works for estimating fees only if you know the specific XCM you want to execute or send.
+
+### Dry Run Call
+
+This API allows to dry-run any extrinsic and obtaint the outcome, if it fails or succeeds and also the local xcm and remote xcm messages sent to other chains.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/dry_run.rs:67:67'
+```
+
+??? interface "Input parameters"
+
+    `origin` ++"OriginCaller"++ <span class="required" markdown>++"required"++</span>
+    
+    The origin used for executing the transaction.
+
+    ---
+
+    `call` ++"Call"++ <span class="required" markdown>++"required"++</span>
+
+    The extrinsic to be executed.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<CallDryRunEffects<Event>, Error>"++
+
+    Effects of dry-running an extrinsic. If an error occurs, it is returned instead of the effects.
+
+    ??? child "Type `CallDryRunEffects<Event>`"
+
+        `execution_result` ++"DispatchResultWithPostInfo"++
+
+        The result of executing the extrinsic.
+
+        ---
+
+        `emitted_events` ++"Vec<Event>"++
+
+        The list of events fired by the extrinsic.
+
+        ---
+
+        `local_xcm` ++"Option<VersionedXcm<()>>"++
+
+        The local XCM that was attempted to be executed, if any.
+
+        ---
+
+        `forwarded_xcms` ++"Vec<(VersionedLocation, Vec<VersionedXcm<()>>)>"++
+
+        The list of XCMs that were queued for sending.
+
+    ??? child "Type `Error`"
+
+        Enum:
+
+        - **`Unimplemented`** - an API part is unsupported
+        - **`VersionedConversionFailed`** - converting a versioned data structure from one version to another failed
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to simulate a cross-chain asset transfer from the Paseo network to the Pop Network using a [reserve transfer](https://wiki.polkadot.network/docs/learn/xcm/journey/transfers-reserve){target=\_blank} mechanism. Instead of executing the actual transfer, the code shows how to test and verify the transaction's behavior through a dry run before performing it on the live network:
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example.js'
+    ```
+
+    !!!note
```

Comment:
Please take this out of the note and add it to the copy above the code sample so the user sees it before they start working with the script. Thank you! 

"This example demonstrates how to simulate a cross-chain asset transfer from the Paseo network to the Pop Network using a [reserve transfer](https://wiki.polkadot.network/docs/learn/xcm/journey/transfers-reserve){target=\_blank} mechanism. Instead of executing the actual transfer, the code shows how to test and verify the transaction's behavior through a dry run before performing it on the live network. 

Replace `INSERT_USER_ADDRESS` with your SS58 address before running the script."